components/event-bus/internal/knative/subscription/controller/subscription/controller.go

```diff
@@ -0,0 +1,49 @@
+package subscription
+
+import (
+	eventingv1alpha1 "github.com/kyma-project/kyma/components/event-bus/api/push/eventing.kyma-project.io/v1alpha1"
+	"github.com/kyma-project/kyma/components/event-bus/internal/knative/subscription/opts"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("subscription-controller")
+
+const (
+	// controllerAgentName is the string used by this controller to identify
+	// itself when creating events.
+	controllerAgentName = "subscription-controller"
+)
+
+// ProvideController returns a Controller that represents the subscription controller. It
+// reconciles only the Kyma Subscription.
+func ProvideController(mgr manager.Manager, opts *opts.Options) (controller.Controller, error) {
+
+	// Setup a new controller to Reconcile Kyma Subscription.
+	r := &reconciler{
+		recorder: mgr.GetRecorder(controllerAgentName),
+		opts: opts,
```

Comment:
the KN-Lib instance should be part of "reconciler" structure. It should not be created at each Reconcile() call. The controller could create it in this place.

Reply:
Done